src/models/apierror.h

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef APIERROR_H
+#define APIERROR_H
+
+#include <QObject>
+#include <QString>
+
+class QByteArray;
+
+class ApiError final {
+  Q_GADGET
+
+ public:
+  ApiError(){};
+  ApiError(const ApiError& other);
+  ApiError& operator=(const ApiError& other);
+
+  // Error codes transcribed from guardian-website/server/lib/errors.ts
+  enum ErrorNumbers : int {
+    NoError = 0,
+
+    MissingPubkeyError = 100,
+    MissingNameError = 101,
+    InvalidPubkeyError = 102,
+    PubkeyAlreadyUsed = 103,
+    KeyLimitReachedError = 104,
+    PubkeyNotFound = 105,
+    SecretKeyNotFound = 106,
+
+    FxaUnauthorizedForRouteError = 110,
+
+    InvalidToken = 120,
+    UserNotFound = 121,
+    DeviceNotFound = 122,
+    NoActiveSubsription = 123,
+    MissingAuthCodeError = 127,
+    MissingAuthCodeVerifierError = 128,
+    AuthCodeExpiredError = 129,
+    InvalidAuthCodeVerifierError = 130,
+    MissingReceiptError = 141,
+    UserAlreadyHasActiveSubscription = 143,
+    ReceiptIsBeingUsed = 145,
+    InvalidTokenError = 147,
+    InvalidSkuError = 148,
+    MissingTokenError = 149,
+    MissingSkuError = 150,
+    FxaTokenValidationError = 151,
+    FxaPlansError = 152,
+    FxaSupportTicketError = 153,
+    FxaSupportTicketNotReturned = 154,
+    SupportTicketIssueTextSizeError = 155,
+    SupportTicketMissingLogsError = 157,
```

Comment:
Missing 156 (`SupportTicketMissingIssueTextError`) and 160 (`AdjustMissingOrInvalidMethod`)